src/fprime/fbuild/builder.py

```diff
@@ -399,16 +386,35 @@ def split_pair(item):
             default_options_text = self.get_settings("default_cmake_options", None)
             default_options = default_options_text.split("\n")
 
-            default_cmake_args = {
+            default_cmake_options = {
                 option: value
                 for (option, value) in [split_pair(item) for item in default_options]
                 if option != ""
             }
 
+            cmake_args.update(default_cmake_options)  # default_cmake_options
+            cmake_args.update(user_cmake_args)  # User-supplied values from command line
+            cmake_args.update(self.get_cmake_args())  # FPRIME_* values (settings.ini)
+
+            # When the new v3 autocoder directory exists, this means we can use the new UT api and preserve the build type
+            v3_autocoder_directory = Path(
+                cmake_args.get("FPRIME_FRAMEWORK_PATH") / "cmake" / "autocoder"
+            )
+            if (
+                v3_autocoder_directory.exists()
+                and self.build_type == BuildType.BUILD_TESTING
+            ):
+                cmake_args["BUILD_TESTING"] = "ON"
+                cmake_args["CMAKE_BUILD_TYPE"] = user_cmake_args.get(
+                    "CMAKE_BUILD_TYPE", "Debug"
+                )
```

Comment:
@thomas-bc  I originally opened this pull request to give the possibility to build UTs using a `CMAKE_BUILD_TYPE` different from `Debug`. I agree with your changes but if I understand correctly the code, you reintroduced back the problem I wanted to remove. 

Am I right or the `CMAKE_BUILD_TYPE` will be always `Debug` if the UTs are built? Is that something that we want by design? I think that the user should have the flexibility to choose the `CMAKE_BUILD_TYPE` in any case but if by design we don't want to give this flexibility then we should document it somewhere because it seems a hidden and unexpected feature.

Reply:
@thomas-bc do you have any comments about this?

Reply:
For newer versions of F´ (v3+) it only defaults to debug if the user has not set "CMAKE_BUILD_TYPE"

Reply:
I tested this and a custom build type does carry into a UT cache!